pkg/multicluster/reconcile.go

```diff
@@ -0,0 +1,26 @@
+package multicluster
+
+import (
+	"context"
+
+	"github.com/solo-io/skv2/pkg/ezkube"
+	"github.com/solo-io/skv2/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type Reconciler interface {
+	// reconcile an object
+	// requeue the object if returning an error, or a non-zero "requeue-after" duration
+	Reconcile(cluster string, object ezkube.Object) (reconcile.Result, error)
+}
+
+type DeletionReconciler interface {
+	// we received a reconcile request for an object that was removed from the cache
+	ReconcileDeletion(cluster string, request reconcile.Request)
+}
```

Comment:
Just wondering out loud why the decision was made to split this interface, and allow the `DeletionReconciler` to be optional. In general I find optional interfaces to be hard to reason about. Instead could we combine them and leave it up to the user whether they want to implement the delete in their code, or just leave it empty.

Reply:
Discussed this offline, reasoning makes sense

Reply:
discussed offline